src/utils/rewriteLink.ts

```diff
@@ -96,7 +96,33 @@ const rewriteLink = (
         )
     }
 
-    // absolute links
+    // absolute links to docs (unversioned)
+    if (href.match(/^https?:\/\/docs.silverstripe.org\/en\/[a-zA-Z]/)) {
+        return createElement(
+            Link,
+            {
+                to: path.join('/', 'en', version, href.replace(/^https?:\/\/docs.silverstripe.org\/en/, '')),
+                className: 'gatsby-link'
+            },
+            domToReact(children, parseOptions)
+        );
+    }
+
+    // absolute links to docs (versioned)
+    const hrefMatch = href.match(/^https?:\/\/docs.silverstripe.org\/en\/([0-9]+)/);
+    if (hrefMatch) {
+        const hrefVersion = hrefMatch[1];
+        return createElement(
+            Link,
+            {
+                to: path.join('/', 'en', hrefVersion, href.replace(/^https?:\/\/docs.silverstripe.org\/en\/([0-9]+)/, '')),
+                className: 'gatsby-link'
+            },
+            domToReact(children, parseOptions)
+        );
+    }
```

Comment:
Technically unrelated change but very similar to the above. Sometimes we explicitly want to link to docs from another version (for example in the 5.0.0 changelog we link to CMS 4 upgrade guides).

This rewrites those links to be relative instead of absolute, and doesn't open them in a new tab.